rust/src/checkpoints.rs

```diff
@@ -100,6 +122,36 @@ pub async fn create_checkpoint_from_table(table: &DeltaTable) -> Result<(), Chec
         &table.table_uri,
     )
     .await?;
+
+    let new_version = table.version + 1;
+
+    if new_version > 0 {
+        let deleted_log_num = cleanup_expired_logs(
+            new_version,
+            table.storage.as_ref(),
+            table.get_state(),
+            &table.table_uri,
+        )
+        .await?;
+        debug!("Deleted {:?} log files.", deleted_log_num);
```

Comment:
Also, instead of always running cleanup, could we propagate the `enableExpiredLogCleanup` delta table config as in https://github.com/delta-io/delta/blob/08b17e0fae5222179a52ee277f93d8ebc0029621/core/src/main/scala/org/apache/spark/sql/delta/DeltaConfig.scala#L306

Reply:
Yea, I think so. Would you like to add here or we can do it in another PR after this?

Reply:
sure, we can do that in a separate PR